hyperspy/signal.py

```diff
@@ -3199,16 +3199,18 @@ def map(self, function,
             parallel=None, **kwargs):
         """Apply a function to the signal data at all the coordinates.
 
-        The function must operate on numpy arrays and the output *must have the
-        same dimensions as the input*. The function is applied to the data at
+        The function must operate on numpy arrays, but the output no longer has
+        to be of the same dimensions. The function is applied to the data at
         each coordinate and the result is stored in the current signal i.e.
-        this method operates *in-place*.  Any extra keyword argument is passed
-        to the function. The keywords can take different values at different
-        coordinates. If the function takes an `axis` or `axes` argument, the
-        function is assumed to be vectorial and the signal axes are assigned to
-        `axis` or `axes`.  Otherwise, the signal is iterated over the
-        navigation axes and a progress bar is displayed to monitor the
-        progress.
+        this method operates *in-place* and overwrites the data. Any extra
+        keyword argument is passed to the function. The keywords can take
+        different values at different coordinates. If the function takes an
+        `axis` or `axes` argument, the function is assumed to be vectorial and
+        the signal axes are assigned to `axis` or `axes`.  Otherwise, the
+        signal is iterated over the navigation axes and a progress bar is
+        displayed to monitor the progress.
+
+        Navigation dimensions and information is always preserved.
```

Comment:
I see. What about "Navigation dimension and axes attributes are always preserved."?